Makefile

```diff
@@ -319,6 +321,7 @@ clean: clean_debuild
 	rm -rf $(BUILD_PYTHON_OUTPUT_DIR)
 	rm -rf $(VENV_DIR)
 	rm -rf $(INSTALL_DEPS_STAMP)
+	rm -rf $(VERSION_FILE) $(SRC_DIR)/__init__.py
```

Comment:
Why can't we keep `$(SRC_DIR)/__init__.py` ?

Reply:
In `clean` target? From my point of view, `clean` should remove all build artifacts, clean up side effects if any (e.g. destroy docker containers created in integration tests) and return working directory to the state before executing build or test targets.

If the question about removing  `$(SRC_DIR)/__init__.py` from git in overall, it also relates to clean working directory. It's better to keep manually created files and generated ones separated to avoid committing unintentional changes. Sometimes generated changed are committed to vcs, but it doesn't seem like a good solution for `__init__.py` and `pyproject.toml`. Otherwise the changes in these files should be included in every commit because every commit affects generated version value in these files.

Reply:
In my opinion, we should not remove `__init__.py`. It's confusing and may be unexpected when you are doing something without using makefile after such a cleaning. Moreover this file might contain some useful code in the future. 
And the version in `pyptoject.toml` remains.
I can suggest replacing versions in this files to default values `0.0.0` in `clean` target, or by pre-commit hook. For me it is more clear.

Reply:
I'm a bit confusing because I'm always thinking that clean build is import property of any software project and that running build or tests from clean working copy shouldn't produce any changes in vcs controlled files.

Though I'm agree that PR doesn't address the issue completely and we also generate changes in pyproject.toml during the build. 
Will close the PR for now as it's incomplete work right now.